From a85193f122b7ec0d377e0a39ac58c64f12c51085 Mon Sep 17 00:00:00 2001
From: Rui Matos <tiagomatos@gmail.com>
Date: Mon, 6 Jan 2014 13:45:00 +0000
Subject: [PATCH] auserver.c: Only write audio data to a file in debug builds

This isn't useful in regular builds and is actually a security
concern.
---
 src/audio/auserver.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/audio/auserver.c b/src/audio/auserver.c
index 29716d3..3e838d0 100644
--- a/src/audio/auserver.c
+++ b/src/audio/auserver.c
@@ -58,9 +58,11 @@ static int play_wave_from_socket(snd_header *header,int audiostream)
     int q,i,n,r;
     unsigned char bytes[CST_AUDIOBUFFSIZE];
     short shorts[CST_AUDIOBUFFSIZE];
+#ifdef DEBUG
     cst_file fff;
 
     fff = cst_fopen("/tmp/awb.wav",CST_OPEN_WRITE|CST_OPEN_BINARY);
+#endif
 
     if ((audio_device = audio_open(header->sample_rate,1,
 				   (header->encoding == CST_SND_SHORT) ?
@@ -107,7 +109,9 @@ static int play_wave_from_socket(snd_header *header,int audiostream)
 	for (q=r; q > 0; q-=n)
 	{
 	    n = audio_write(audio_device,shorts,q);
+#ifdef DEBUG
 	    cst_fwrite(fff,shorts,2,q);
+#endif
 	    if (n <= 0)
 	    {
 		audio_close(audio_device);
@@ -116,7 +120,9 @@ static int play_wave_from_socket(snd_header *header,int audiostream)
 	}
     }
     audio_close(audio_device);
+#ifdef DEBUG
     cst_fclose(fff);
+#endif
 
     return CST_OK_FORMAT;
 
-- 
1.8.3.1

